socs/Lakeshore/Lakeshore336.py

```diff
@@ -0,0 +1,1195 @@
+      
+        if not self.comm.is_open:
+            self.comm.open()
+        
+        out = self.comm.read_until(size = size).strip().decode()
+        self.comm.close()
+
+        time.sleep(delay)
+
+        return out
+
+    def read_size(self, size = 256, delay = None):
+        r'''
+        Reads data from the device attached to port comm. Implemented via pyserial 'read' command: reads until receives size bytes.
+
+        Parameters
+        ------
+        comm : serial.Serial object
+            The port on the local machine attached to the device. See pyserial documentation. Must be configured according to the lakeshore manual.
+        
+        size : int, optional
+            Number of bytes to read. If provided, may be less than full message length.
+        
+        Returns
+        ------
+        out : str
+            Message from lakeshore device.
+        '''       
+        if not delay: delay = self.std_delay
+ 
+        if not self.comm.is_open:
+            self.comm.open()
+        
+        out = self.comm.read(size = size).strip().decode()
+        self.comm.close()
+        
+        time.sleep(delay)
+        
+        return out
+
+    def get_id(self):
+        '''Get identification information of the Lakeshore module'''
+        return self.msg('*IDN?')
+
+    def get_kelvin(self, inp):
+        '''Return a temperature reading of the specified input ('A', 'B', 'C', or 'D')
+        or '0' for all inputs
+
+        Parameters
+        ----------
+        inp : str
+            channel to query
+
+        Returns
+        -------
+        str
+            plain text of temperature reading
+
+        Raises
+        ------
+        ValueError
+            Invalid input channel arguments
+        '''
+        if inp not in ['0', 'A', 'B', 'C', 'D']:
+            raise ValueError(f'invalid input in msg_kelvin: {inp}')
+
+        return self.msg(f'KRDG? {inp}')
+        
+    def get_sensor(self, inp):
+        '''Return a sensor reading of the specified input ('A', 'B', 'C', or 'D')
+        or '0' for all inputs
+
+        Parameters
+        ----------
+        inp : str
+            channel to query
+
+        Returns
+        -------
+        str
+            plain text of temperature reading
+
+        Raises
+        ------
+        ValueError
+            Invalid input channel arguments
+        '''     
+        if inp not in ['0', 'A', 'B', 'C', 'D']:
+            raise ValueError(f'invalid input in msg_sensor: {inp}')
+
+        return self.msg(f'SRDG? {inp}')
+
+    def get_heater_range(self, htr):
+        return self.heaters[htr].get_heater_range()
+
+    def get_max_current(self, htr):
+        return self.heaters[htr].get_max_current()
+
+    def get_heater_percent(self, htr):
+        return self.heaters[htr].get_heater_percent()
+
+    def get_setpoint(self, htr):
+        return self.heaters[htr].get_setpoint()
+
+class Channel:
+
+    def __init__(self, ls, _input):
```

Comment:
I think you might be trying to avoid the built-in function `input` here? Convention is to put an underscore after the variable, a leading underscore would imply a private variable. Would change this to `input_`

Reply:
yes, fixed and fixed all others too